### PR TITLE
[SCM-947]

### PR DESCRIPTION
**Pull request by cquoss**
Pull request for SCM-947.
Make all parameter of changelog request available in changelog mojo.

### CODE REVIEW

**Review by newblaque — APPROVED** [review]

**Comment by cquoss** [comment]
Something in Travis CI seems broken, unrelated to my changes. For builds with Java 6 and 7 the config does not seem to fit any more.

**Comment by cquoss** [comment]
Another point: Where lies the documentation for the changelog goal of the maven-scm-plugin? An update here should be part of / done in parallel by me / for this pull request.
Please guide me.

File: maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java [hunk 1]
```diff
@@ -125,24 +145,58 @@ public void execute()
...
+            
+            if ( StringUtils.isNotEmpty( endScmVersion ) )
+            {
+                ScmVersion endRev =
+                    getScmVersion( StringUtils.isEmpty( endScmVersionType ) ? VERSION_TYPE_REVISION 
+                        : endScmVersionType, endScmVersion );
+                request.setEndRevision( endRev );
+            }
+
+            request.setLimit(limit);
```
Comment (michael-o) [inline maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java:165]: spaces missing

File: maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java [hunk 2]
```diff
@@ -125,24 +145,58 @@ public void execute()
...
+            
+            if ( StringUtils.isNotEmpty( startDate ) )
             {
-                result = provider.changeLog( repository, getFileSet(), this.parseDate( localFormat, this.startDate ),
-                                             this.parseDate( localFormat, this.endDate ), 0,
-                                             (ScmBranch) getScmVersion( scmVersionType, scmVersion ), dateFormat );
+                request.setStartDate( parseDate( localFormat, startDate ) );
             }
+
+            if ( StringUtils.isNotEmpty( startScmVersion ) )
```
Comment (michael-o) [inline maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java:190]: Why has this moved down here?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java (hunk 2 above)
Comment (cquoss) [reply maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java:190]: Hmm. I believe i did some sort of alphabetical ordering.

Comment (cquoss) [reply maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java:190]: Should this be moved to somewhere else?

File: maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java (hunk 1 above)
Comment (cquoss) [reply maven-scm-plugin/src/main/java/org/apache/maven/scm/plugin/ChangeLogMojo.java:165]: Changed and committed / pushed.

**Comment by michael-o** [comment]
Does this require SCM-948 first?

**Comment by cquoss** [comment]
TBH: I do not know any more. Let us do this: first merge this PR because this should introduce log request all the way down to the command line calls of svn and git. And then SCM-947 should be easy-peasy simply enabling the limit param on top of this.
I will check this one and post an update here.
Can you resolve the conversation above or reply to my comments, please?

**Comment by cquoss** [comment]
OK. I have been too optimistic. But at least having this will help test SCM-948 since the mojo then sends down a changelog request. But it still uses the deprecated methods with single parms instead of the changelog request. This needs to be changed with SCM-948. But SCM-947 could be merged from my point of view. -Dlimit=1 on changelog goal shows no effect as one can easily verify. But that is up to SCM-948 to fix. And documentation of Changelog Goal of Mojo needs update. Where does that reside?

**Comment by michael-o** [comment]
I have merge SCM-948, am I good to review this now?

**Comment by cquoss** [comment]
Yes. You are.

**Comment by jira-importer** [comment]
Resolve #1172

**Pull request closed without merging**